crates/scheduler/src/consumer.rs

```diff
@@ -34,10 +30,10 @@ pub fn valid_business_statuses() -> Vec<&'static str> {
 }
 
 #[instrument(skip_all)]
-pub async fn start_consumer(
-    state: &AppState,
-    settings: sync::Arc<settings::SchedulerSettings>,
-    workflow_selector: workflows::WorkflowSelectorFn,
+pub async fn start_consumer<T: SchedulerAppState + Send + Sync + Clone + 'static>(
```

Comment:
I think the `state` won't have `'static` lifetime. As the construction will happen at runtime. Still we might need to test this out